zombies.js

```diff
@@ -158,8 +196,22 @@
  * @name equip
  * @param {Weapon} itemToEquip  The weapon item to equip.
  */
-
-
+Player.prototype.equip = function(itemToEquip) {
+  let index = this.getPack().indexOf(itemToEquip); // index of the item in the pack
+  let instance = itemToEquip instanceof Weapon; // item is a weapon
+  if (instance && index >= 0) {
+    // if item is in my bag and is a weapon
+    if (this.equipped === false) {
+      this.discardItem(itemToEquip); // take the item I want to equip out of the pack
+      this.equipped = itemToEquip; // equip the item I want to equip
+    } else {
+      // if I already have an item equipped
+      this.takeItem(this.equipped); // put the equipped item in the pack
+      this.discardItem(itemToEquip); // take the item I want to equip out of the pack
```

Comment:
The takeItem and the discardItem should be switched. You want to take item out of the pack, then replace it with what's already equipped.